### PR TITLE
add probability distribution over predicted cells

### PR DESCRIPTION
**Pull request by tzufgoogle**
Add a probability over predicted cells according to their distance from start point

### CODE REVIEW

File: cabby/model/text/dual_encoder/dataset_rvs.py [hunk 1]
```diff
@@ -22,13 +22,20 @@
 import torch
 from transformers import DistilBertTokenizerFast
 
+import sys
+sys.path.append("/home/tzuf_google_com/dev/cabby")
```
Comment (jasonbaldridge) [inline cabby/model/text/dual_encoder/dataset_rvs.py:26]: Remove developer specific information.

File: cabby/model/text/dual_encoder/dataset_rvs.py [hunk 2]
```diff
@@ -45,16 +52,29 @@ class TextGeoSplit(torch.utils.data.Dataset):
   def __init__(self, data: pd.DataFrame, s2level: int, 
     unique_cells_df: pd.DataFrame, cellid_to_label: Dict[int, int]):
     # Tokenize instructions.
+
```
Comment (jasonbaldridge) [inline cabby/model/text/dual_encoder/dataset_rvs.py:55]: Remove extra blank line.

File: cabby/model/text/dual_encoder/train.py [hunk 3]
```diff
@@ -88,6 +88,8 @@ def evaluate(self, validation_set: bool = True):
         cellids = batch['cellid'].float().to(self.device)
         neighbor_cells = batch['neighbor_cells'].float().to(self.device) 
         far_cells = batch['far_cells'].float().to(self.device)
+        distibution = batch['distribution'].to(self.device)
```
Comment (jasonbaldridge) [inline cabby/model/text/dual_encoder/train.py:91]: sp: distribution

File: cabby/model/text/dual_encoder/train.py [hunk 4]
```diff
@@ -99,6 +101,7 @@ def evaluate(self, validation_set: bool = True):
           batch_dim, cell_dim, output_dim)
         text_embedding_exp = text_embedding.unsqueeze(1)
         output = self.cos(cellid_embedding_exp, text_embedding_exp)
+        # output = output * distibution
```
Comment (jasonbaldridge) [inline cabby/model/text/dual_encoder/train.py:104]: Remove commented line.

File: cabby/model/text/dual_encoder/dataset_rvs.py [hunk 5]
```diff
@@ -45,16 +52,29 @@ class TextGeoSplit(torch.utils.data.Dataset):
...
+
+    start_points = data.start_point.apply(
+      lambda x: gutil.point_from_list_coord(x))
+    
+    distances = start_points.apply(
+      lambda start_p: unique_cells_df.point.apply(
+        lambda end_p: gutil.get_distance_between_points(start_p, end_p)))
+
+    self.start_ditribution = distances.apply(
```
Comment (jasonbaldridge) [inline cabby/model/text/dual_encoder/dataset_rvs.py:64]: sp: start_distribution

File: cabby/model/text/dual_encoder/train.py (hunk 4 above)
Comment (jasonbaldridge) [reply cabby/model/text/dual_encoder/train.py:104]: Or, perhaps this should have an if statement that multplies by the distance distribution if desired?

File: cabby/model/text/dual_encoder/train.py (hunk 3 above)
Comment (jasonbaldridge) [reply cabby/model/text/dual_encoder/train.py:91]: Also, should this be distance_distribution?

**Review by jasonbaldridge — CHANGES_REQUESTED** [review]
Some things to fix / update!

File: cabby/model/text/dual_encoder/dataset_rvs.py (hunk 1 above)
Comment (tzufgoogle) [reply cabby/model/text/dual_encoder/dataset_rvs.py:26]: Removed

File: cabby/model/text/dual_encoder/dataset_rvs.py (hunk 2 above)
Comment (tzufgoogle) [reply cabby/model/text/dual_encoder/dataset_rvs.py:55]: Removed

File: cabby/model/text/dual_encoder/dataset_rvs.py (hunk 5 above)
Comment (tzufgoogle) [reply cabby/model/text/dual_encoder/dataset_rvs.py:64]: Changed to start_distribution

File: cabby/model/text/dual_encoder/train.py (hunk 3 above)
Comment (tzufgoogle) [reply cabby/model/text/dual_encoder/train.py:91]: changed to distance_distribution

File: cabby/model/text/dual_encoder/train.py (hunk 4 above)
Comment (tzufgoogle) [reply cabby/model/text/dual_encoder/train.py:104]: Added a condition statement

**Review by tzufgoogle — COMMENTED** [review]
Added the distance distribution as an option

**Review by jasonbaldridge — APPROVED** [review]
LGTM

**Pull request merged**